tljh/normalize.py

```diff
@@ -8,14 +8,15 @@ def generate_system_username(username):
     """
     Generate a posix username from given username.
 
-    If username < 26 char, we just return it.
+    If username < 26 char, we just return it
+    (with any spaces replaced by dashes).
     Else, we hash the username, truncate username at
     26 char, append a '-' and first add 5char of hash.
     This makes sure our usernames are always under 32char.
     """
 
     if len(username) < 26:
-        return username
+        return username.replace(' ', '-')
```

Comment:
This results in a clash between `a-b c` and `a b-c`

Reply:
Good point. So maybe it's better to remove any illegal characters and then append a hash of the original name, as we do for long usernames ...